pkg/lib/machine-info.es6

```diff
@@ -191,3 +191,46 @@ export function udev_info(address) {
     }
     return pr;
 }
+
+const memoryRE = /^(\w+[ \w+]*):\s(.*)/;
+
+function parseMemoryInfo(text) {
+    var info = {};
+    text.split("\n\n").map(paragraph => {
+        let locator = null;
+        let props = {};
+
+        paragraph = paragraph.trim();
+        if (!paragraph)
+            return;
+
+        paragraph.split("\n").map(line => {
+            line = line.trim();
+            let match = line.match(memoryRE);
+            if (match)
+              props[match[1]] = match[2];
+        });
+
+        locator = props["Locator"];
+        if (locator)
+            info[locator] = props;
+    });
+    return info;
+}
+
+var memory_info_promises = { };
+
+export function memory_info(address) {
```

Comment:
This could get a comment that documents what kind/structure of information gets returned here. I know the other existing functions don't do that, we should fix that.

Reply:
Well, the library should document it for its callers. The callers should not document the library functions that they call :) `processMemory()` does not call `memory_info()`, so you wouldn't even find it with grepping. Also, that comment describes a list, while this function returns a `locator → properties` map.

Reply:
Marking this as resolved as comment is now present.